toucan_connectors/toucan_connector.py

```diff
@@ -36,3 +36,7 @@ def __init_subclass__(cls):
     @abstractmethod
     def get_df(self, data_source: ToucanDataSource) -> pd.DataFrame:
         """Main method to retrieve a pandas dataframe"""
+
+    def get_status(self) -> str:
```

Comment:
I think we should discuss this, to me me a str is not going to be very useful, I think we should return an array of error objects such as pseudo code:  "[(true, I can speak to localhost:port), (false, I cannot query it)]" or "[(false, I cannot speak to localhost:port), (false, cannot query it)]"